functions/index.js

```diff
@@ -72,72 +73,171 @@ exports.resetVotesScheduledFunction = functions.region('europe-west1').pubsub.sc
+                            var section_power_up_funds = PowerUpSnapshot.child("funds").child(section).val()
+
+                            if(section_power_up_funds >= power_up_cost){
+                                val_section *= 2
+                            }
+                        }
+
+                        if(val_section > max){
+                            max = val_section
+                            new_owner = section
+                        } else if(val_section > second_max){
+                            second_max = val_section
+                        }
                     }
                 }
-            }
-        
+            
 
-            // no capture if equality
-            if(max != second_max){
+                // no capture if equality
+                if(max != second_max){
 
-                sectionsScores.set(new_owner, sectionsScores.get(new_owner) + 1)
-    
-                refZones.child(zone_name).child("owner").set(new_owner, set_error_callback)
-            } else {
+                    sectionsScores.set(new_owner, sectionsScores.get(new_owner) + 1)
+                
+                    refZones.child(zone_name).child("owner").set(new_owner, set_error_callback)
+                } else {
+
+                    sectionsScores.set(current_owner, sectionsScores.get(current_owner) + 1)
+
+                }
+
+            })
+
+            sectionsScores.forEach( (val, key) => {
+                if(key != "NONE"){
+                    refSections.child(key).child("score").set(val, set_error_callback)
+                }
+            })
+
+            //remove money of used power ups
+            const refPowerUp = db.ref('PowerUp/SuperBigMaxPower')
+            refPowerUp.once('value').then( (PowerUpSnapshot) => {
+            
+                var power_up_cost = PowerUpSnapshot.child("value").val()
+            
+                PowerUpSnapshot.child("funds").forEach( (fund_section_snapshot) => {
+                
+                    var section = fund_section_snapshot.key
+                
+                    if(fund_section_snapshot.val() >= power_up_cost){
+                    
+                        refPowerUp.child("funds").child(section).set(admin.database.ServerValue.increment(-power_up_cost), set_error_callback)
+                    }
+                })
+            })
+
+
+        })
+    })
+
+    return null
+})
+
+exports.giveMoneyScheduledFunction = functions.region('europe-west1').pubsub.schedule("0 12 * * *").onRun((context) => {
+
+    console.log("It's noon!")
+  
+    const db = getDatabase()
+    const refUsers = db.ref('Users')
+
+    //take snapshot of Users
+    refUsers.once('value').then( (usersSnapshot) => {
 
-                sectionsScores.set(current_owner, sectionsScores.get(current_owner) + 1)
+        //count number of user for each section
+        const sectionsUserCount = new Map()
+        for(var i = 0, size = sections.length; i < size ; i++){
```

Comment:
Could collapse this loop with the one below

Reply:
I think bc the second one add 1 to an existing value, we need to initalize the values with either a loop before or a condition in the second one (both are 3 loc)